### PR TITLE
controllers: tiltfile reconciler

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas, @landism,

Please review the following commits I made in branch nicks/tiltfile2:

4fe7f7bf31b98012b518000ee7a8751edb1ed19e (2021-07-30 17:58:50 -0400)
controllers: tiltfile reconciler

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/controllers/core/tiltfile/reconciler.go [hunk 1]
```diff
@@ -63,16 +79,178 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
...
+		ctx = run.entry.WithLogger(ctx, r.st)
+	}
+
+	// Check to see if we have a BuildEntry triggering this tiltfile.
+	//
+	// This is a short-term hack to make this interoperate with the legacy subscriber.
+	// The "right" version would read from FileWatch and other dependencies to determine
+	// when to build.
+	be := r.buildSource.Entry()
+	if be != nil && be.Name.String() == nn.Name && (step == runStepNone || step == runStepDone) {
```
Comment (milas) [inline internal/controllers/core/tiltfile/reconciler.go:109]: Why does this trigger on `runStepDone`?

File: internal/controllers/core/tiltfile/reconciler.go [hunk 2]
```diff
@@ -89,3 +267,77 @@ func indexTiltfile(obj client.Object) []indexer.Key {
...
+
+	return false
+}
+
+// Represent the steps of Tiltfile execution.
+type runStep int
+
+const (
+	// Nothing needs to be done.
+	runStepNone runStep = iota
```
Comment (milas) [inline internal/controllers/core/tiltfile/reconciler.go:292]: Is this more accurately `// Tiltfile execution is waiting to start`?

File: internal/engine/configs/configs_controller_test.go [hunk 3]
```diff
@@ -219,6 +233,29 @@ func (f *ccFixture) setManifestResult(m model.Manifest) {
...
+}
+
+// Wait for the next item on the workqueue, then run reconcile on it.
+func (f *ccFixture) popQueue() {
+	f.T().Helper()
+
+	done := make(chan struct{})
+	go func() {
+		item, _ := f.q.Get()
+		_, _ = f.tfr.Reconcile(f.ctx, item.(reconcile.Request))
```
Comment (milas) [inline internal/engine/configs/configs_controller_test.go:247]: Would it be practical to assert on no errors here?

**Review by milas — APPROVED** [review]

File: internal/controllers/core/tiltfile/reconciler.go (hunk 1 above)
Comment (nicks) [reply internal/controllers/core/tiltfile/reconciler.go:109]: like if the tiltfile restarts! will add a comment on this

File: internal/controllers/core/tiltfile/reconciler.go (hunk 2 above)
Comment (nicks) [reply internal/controllers/core/tiltfile/reconciler.go:292]: done

File: internal/engine/configs/configs_controller_test.go (hunk 3 above)
Comment (nicks) [reply internal/engine/configs/configs_controller_test.go]: good idea! done

**Pull request merged**